constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +506,369 @@ func TestRegoRewriterErrorCases(t *testing.T) {
+			rr, err := New(NewPackagePrefixer(tc.prefix), []string{"data.lib"}, []string{"data.ext"})
+			if err != nil {
+				t.Fatalf("Failed to create RegoRewriter %s", err)
+			}
+			if err := rr.AddEntryPoint("path", tc.content); err != nil {
+				if !errors.Is(err, tc.wantError) {
+					t.Fatalf("failed to add base source %q", err)
+				}
+				return
+			}
+			sources, err := rr.Rewrite()
+			if !errors.Is(err, tc.wantError) {
+				t.Errorf("Rewrite() got error = %v, want %v", err, tc.wantError)
+			}
+
+			result, err := sources.AsMap()
+			if err != nil {
+				t.Fatalf("unexpected error during Sources.AsMap: %s", err)
+			}
+
+			if diff := cmp.Diff(tc.wantResult, result); diff != "" {
+				t.Errorf("result differs from desired:\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestRegoRewriteLib(t *testing.T) {
+	tests := []struct {
+		name       string
+		prefix     string
+		content    string
+		wantResult map[string]string
+		wantError  error
+	}{
+		{
+			name:   "entry point imports lib",
+			prefix: "prefix.path",
+			content: `package lib.alpha
+import data.lib.alpha
+check(objects) = object {
+	object := objects[_]
+	beta.check(object)
+	data.lib.beta.check(object)
+}`,
+			wantResult: map[string]string{"path": `package prefix.path.lib.alpha
+
+import data.prefix.path.lib.alpha
+
+check(objects) = object {
+	object := objects[_]
+	beta.check(object)
+	data.prefix.path.lib.beta.check(object)
+}
+`},
+		}, {
+			name:   "entry point binds data.lib to var",
+			prefix: "prefix",
+			content: `package lib.mylib
+myfunc() {
+	x := data.lib
+ y := x[_]
+}`,
+			wantResult: map[string]string{"path": `package prefix.lib.mylib
+
+myfunc {
+	x := data.prefix.lib
+	y := x[_]
+}
+`},
+		}, {
+			name:   "lib uses data.lib[_]",
+			prefix: "prefix",
+			content: `package lib.alpha
+check(object) {
+  x := data.lib[_]
+  object == "foo"
+}`,
+			wantResult: map[string]string{"path": `package prefix.lib.alpha
+
+check(object) {
+	x := data.prefix.lib[_]
+	object == "foo"
+}
+`},
+		}, {
+			name:   "lib references input",
+			prefix: "prefix",
+			content: `package lib.myLib
+is_foo(name) {
+	input.foo[name]
+}`,
+			wantResult: map[string]string{"path": `package prefix.lib.myLib
+
+is_foo(name) {
+	input.foo[name]
+}
+`},
+		}, {
+			name:   "walk data.lib",
+			prefix: "prefix",
+			content: `package lib.myLib
+is_foo(name) {
+  walk(data.lib, [p, v])
+}`,
+			wantResult: map[string]string{"path": `package prefix.lib.myLib
+
+is_foo(name) {
+	walk(data.prefix.lib, [p, v])
+}
+`},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			rr, err := New(NewPackagePrefixer(tc.prefix), []string{"data.lib"}, []string{"data.ext"})
+			if err != nil {
+				t.Fatalf("Failed to create RegoRewriter %s", err)
+			}
+			if err := rr.AddLib("path", tc.content); err != nil {
+				if !errors.Is(err, tc.wantError) {
+					t.Fatalf("failed to add lib source %q", err)
+				}
+				return
+			}
+			sources, err := rr.Rewrite()
```

Comment:
Can `Rewrite()` ever fail? I don't see any tests that exercise this path. Either add a test that exercises that path, or if it can't reasonably fail then remove the handling logic.

Reply:
Added two Invalid lib test cases from `TestRegoRewriter`